### DISCUSSION TITLE
Base table or view not found: 1146 Table 'sos.role_user' doesn't exist

### DISCUSSION THREAD

**Discussion by zaheer48** [spatie/laravel-permission#2743] [Q&A]
I want to impliment role base access to users in api. user sending access_token in api. but it is giving following error when i use 
middleware like this 
Route::group(['middleware' => ['role:super-admin']], function () {
}
Base table or view not found: 1146 Table 'sos.role_user' doesn't exist

**Comment by drbyte** [comment]
That probably means you've created a `roles()` function in your User model, and that is overriding the method provided by the `HasRoles` trait.  
Or you haven't implemented the `HasRoles` trait.

Both are explained in the documentation.
